common/src/main/java/com/microsoft/identity/common/adal/internal/net/WebRequestHandler.java

```diff
@@ -0,0 +1,100 @@
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.identity.common.adal.internal.net;
+
+import android.os.Build;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.HeaderField;
+import static com.microsoft.identity.common.adal.internal.AuthenticationConstants.MediaType;
+
+/**
+ * It uses one time async task. WebRequest are wrapped here to prevent multiple
+ * reuses for same tasks. Each request returns a handler for cancel action. Call
+ * this from UI thread to correctly create async task and execute.
+ */
+public class WebRequestHandler implements IWebRequestHandler {
+
+    private static final String TAG = "WebRequestHandler";
+
+    /**
+     * Header for accept.
+     */
+    public static final String HEADER_ACCEPT = HeaderField.ACCEPT;
+
+    /**
+     * Header for json type.
+     */
+    public static final String HEADER_ACCEPT_JSON = MediaType.APPLICATION_JSON;
+
+    private UUID mRequestCorrelationId = null;
+
+    @Override
+    public HttpWebResponse sendGet(URL url, Map<String, String> headers) throws IOException {
+        final HttpWebRequest request = new HttpWebRequest(url, HttpWebRequest.REQUEST_METHOD_GET, updateHeaders(headers));
+        return request.send();
+    }
+
+    @Override
+    public HttpWebResponse sendPost(URL url, Map<String, String> headers, byte[] content,
+                                    String contentType) throws IOException {
+        final HttpWebRequest request = new HttpWebRequest(
+                url,
+                HttpWebRequest.REQUEST_METHOD_POST,
+                updateHeaders(headers),
+                content,
+                contentType);
+        return request.send();
+    }
+
+    private Map<String, String> updateHeaders(final Map<String, String> headers) {
+
+        if (mRequestCorrelationId != null) {
+            headers.put(AuthenticationConstants.AAD.CLIENT_REQUEST_ID, mRequestCorrelationId.toString());
+        }
+
+        headers.put(AuthenticationConstants.AAD.ADAL_ID_PLATFORM, "Android");
+        // TODO don't make this dependency circular
+        //headers.put(AuthenticationConstants.AAD.ADAL_ID_VERSION, AuthenticationContext.getVersionName());
```

Comment:
Doesn't make sense to send specific header info for ADAL in shared code - does this need to get injected elsewhere? (probably)

Reply:
I believe this was rudimentary telemetry being recorded by EVO.  We'll need to figure out how a design that allows us to inject these values as part of every request.